docs/reference/extensions/go-framework-extension.rst

```diff
@@ -48,6 +48,28 @@ charm can set it by running ``juju config <application> token=<token>``.
           description: The token for the service.
           type: string
 
+In addition to this, you can set the configuration option to be
+mandatory by setting the ``optional`` key to ``false``. This will
+block the charm and stop services until the configuration is supplied. For example,
+if your application needs an ``api-token`` to function correctly you can set
+``optional``, as shown below. This will block the charm and stop the
+services until the ``api-token`` configuration is supplied.
+
+.. code-block:: yaml
+    :caption: charmcraft.yaml
+
+    config:
+      options:
+        api-token:
+          description: The token necessary for the service to run.
+          type: string
+          optional: false
+
+.. note::
+
+    A configuration with ``optional: false`` can't also have a ``default`` key.
+    If it has both, the charm will fail to pack.
```

Comment:
Could we make this a reusable block? Like: https://github.com/canonical/charmcraft/blob/b4aee1e97868997e878fcad42109740f536f044a/docs/tutorial/write-your-first-kubernetes-charm-for-a-django-app.rst?plain=1#L59